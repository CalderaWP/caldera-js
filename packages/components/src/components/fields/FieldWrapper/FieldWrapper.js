import { fieldWrapperClassNames, parseAttributes } from '../util';
import PropTypes from 'prop-types';
import React from 'react';

/**
 *
 * @param fieldType
 * @param attributes
 * @param children
 * @return {*}
 * @constructor
 */
export const FieldWrapper = ({ fieldType, attributes, children }) => {
	return (
		<div className={fieldWrapperClassNames(fieldType)} {...attributes}>
			{children}
		</div>
	);
};

FieldWrapper.propTypes = {
	fieldType: PropTypes.string.isRequired,
	children: PropTypes.oneOfType([
		PropTypes.arrayOf(PropTypes.node),
		PropTypes.node
	]).isRequired,
	attributes: PropTypes.object
};
