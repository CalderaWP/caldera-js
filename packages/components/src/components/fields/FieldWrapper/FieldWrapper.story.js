import React from 'react';
import { storiesOf } from '@storybook/react';
import { FieldWrapper } from './FieldWrapper';
import { InputField } from '../InputField/InputField';
const onChange = newValue => {};
storiesOf('FieldWrapper', module).add('Wraps an input', () => (
	<FieldWrapper fieldType={'text'}>
		<input />
	</FieldWrapper>
));

storiesOf('FieldWrapper', module).add('With checkboxes inside', () => (
	<FieldWrapper fieldType={'text'}>
		<InputField
			label={'Not Checked'}
			html5type={'checkbox'}
			value={false}
			onChange={onChange}
		/>
		<InputField
			label={'Checked'}
			html5type={'checkbox'}
			value={true}
			onChange={onChange}
		/>
	</FieldWrapper>
));
