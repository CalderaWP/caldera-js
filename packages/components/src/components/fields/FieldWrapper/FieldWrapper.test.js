import * as React from "react";
import { cleanup, render } from "react-testing-library";
import { shallow } from "enzyme";

import { FieldWrapper } from "./FieldWrapper";

describe("Field wrapper", () => {
	afterEach(cleanup);

	it("Matches snapshot", () => {
		const component = render(
			<FieldWrapper fieldType={"checkbox"} legend={"Check All options"}>
				<label htmlFor="one">One</label>
				<input id="one" type={"checkbox"} />
				<label htmlFor="two">Two</label>
				<input id="two" type={"checkbox"} />
			</FieldWrapper>
		);
		expect(component).toMatchSnapshot();
	});

	it.skip("applies an array of classnames", () => {
		const component = shallow(
			<FieldWrapper
				fieldType={"checkbox"}
				className={["strange", "has-error"]}
			>
				found
			</FieldWrapper>
		);
		expect(component.hasClass("has-error")).toBe(true);
		expect(component.hasClass("strange")).toBe(true);
	});

	it.skip("applies an string of classnames", () => {
		const component = shallow(
			<FieldWrapper fieldType={"checkbox"} className={"strange"}>
				found
			</FieldWrapper>
		);
		expect(component.hasClass("strange")).toBe(true);
	});
});
