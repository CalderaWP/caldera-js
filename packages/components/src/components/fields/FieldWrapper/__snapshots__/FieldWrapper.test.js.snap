// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Field label Matches snapshot 1`] = `
<div
  className="caldera-field-wrapper caldera-field-wrapper-checkbox"
>
  <label
    htmlFor="one"
  >
    One
  </label>
  <input
    id="one"
    type="checkbox"
  />
  <label
    htmlFor="two"
  >
    Two
  </label>
  <input
    id="two"
    type="checkbox"
  />
</div>
`;
